### PR TITLE
CPPCheck Cleanup

### PR DESCRIPTION
**Pull request by Lemongrass3110**
Fixes a lot of CPPCheck warnings

Added some TODO comments

### CODE REVIEW

File: src/map/npc.c [hunk 1]
```diff
@@ -2303,15 +2303,16 @@ struct npc_data* npc_add_warp(char* name, short from_mapid, short from_x, short
  */
 static const char* npc_parse_warp(char* w1, char* w2, char* w3, char* w4, const char* start, const char* buffer, const char* filepath)
 {
-	int x, y, xs, ys, to_x, to_y, m;
+	int m;
+	short x, y, xs, ys, to_x, to_y;
 	unsigned short i;
-	char mapname[32], to_mapname[32];
+	char mapname[32], to_mapname[32]; // TODO: Check why this does not use MAP_NAME_LENGTH_EXT
```
Comment (cydh) [inline src/map/npc.c]: change it then

File: src/map/npc.c [hunk 2]
```diff
@@ -2397,8 +2400,8 @@ static const char* npc_parse_shop(char* w1, char* w2, char* w3, char* w4, const
 	}
 	else
 	{// w1=<map name>,<x>,<y>,<facing>
-		char mapname[32];
-		if( sscanf(w1, "%31[^,],%d,%d,%d", mapname, &x, &y, &dir) != 4
+		char mapname[32];  // TODO: Check why this does not use MAP_NAME_LENGTH_EXT
```
Comment (cydh) [inline src/map/npc.c]: change it then

File: src/map/npc.c [hunk 3]
```diff
@@ -2702,9 +2706,9 @@ static const char* npc_parse_script(char* w1, char* w2, char* w3, char* w4, cons
 	}
 	else
 	{// npc in a map
-		char mapname[32];
+		char mapname[32]; // TODO: Check why this does not use MAP_NAME_LENGTH_EXT
```
Comment (cydh) [inline src/map/npc.c]: change it then

File: src/map/npc.c [hunk 4]
```diff
@@ -2859,9 +2864,9 @@ const char* npc_parse_duplicate(char* w1, char* w2, char* w3, char* w4, const ch
 		x = y = dir = 0;
 		m = -1;
 	} else {
-		char mapname[32];
+		char mapname[32]; // TODO: Check why this does not use MAP_NAME_LENGTH_EXT
```
Comment (cydh) [inline src/map/npc.c]: change it then

File: src/map/npc.c [hunk 5]
```diff
@@ -3299,9 +3304,9 @@ void npc_parse_mob2(struct spawn_data* mob)
 
 static const char* npc_parse_mob(char* w1, char* w2, char* w3, char* w4, const char* start, const char* buffer, const char* filepath)
 {
-	int num, class_, m,x,y,xs,ys, i,j;
-	int mob_lv = -1, ai = -1, size = -1;
-	char mapname[32], mobname[NAME_LENGTH];
+	int num, class_, i, j, mob_lv = -1, ai = -1, size = -1;
+	short m,x,y,xs,ys;
+	char mapname[32], mobname[NAME_LENGTH]; // TODO: Check why this does not use MAP_NAME_LENGTH_EXT
```
Comment (cydh) [inline src/map/npc.c]: change it then

File: src/map/npc.c [hunk 6]
```diff
@@ -3495,15 +3500,15 @@ static const char* npc_parse_mapflag(char* w1, char* w2, char* w3, char* w4, con
 		state = 0;	//Disable mapflag rather than enable it. [Skotlex]
 
 	if (!strcmpi(w3, "nosave")) {
-		char savemap[32];
-		int savex, savey;
+		char savemap[32]; // TODO: Check why this does not use MAP_NAME_LENGTH_EXT
```
Comment (cydh) [inline src/map/npc.c]: change it then

**Comment by Lemongrass3110** [comment]
@cydh copy paste comment? xD

**Comment by cydh** [comment]
@Lemongrass3110 yes, and bored in the mid of copy-paste

**Comment by Lemongrass3110** [comment]
Any progress on this? I dont really know which max value should be used exactly.

**Comment by cydh** [comment]
that max value, sounds good, no report about map-crash because of it. :P
ask @aleos89 to merge this. he's mastering merge PR with "__We can’t automatically merge this pull request.__"

**Comment by aleos89** [comment]
I can look back at this. I'm assuming that the value should be 20 as well for MAX_BG_USERS. I haven't seen any documentation stating that the sizes are larger than that.

**Comment by Akkarinage** [comment]
Anyone fancy merging this? Providing it's not too old and will not break something. Why do we leave PRs open this long?

**Comment by aleos89** [comment]
I believe this was left open because no one confirmed the max players for Battlegrounds.

**Comment by aleos89** [comment]
Eh, I merged this horribly wrong. Sorry for the mass combination of stuff.

File: src/map/battle.c [hunk 7]
```diff
@@ -320,7 +320,8 @@ int battle_delay_damage(unsigned int tick, int amotion, struct block_list *src,
 	if (sc && sc->data[SC_DEVOTION] && sc->data[SC_DEVOTION]->val1)
 		d_tbl = map_id2bl(sc->data[SC_DEVOTION]->val1);
 
-	if( d_tbl && sc && sc->data[SC_DEVOTION] && damage > 0 && skill_id != PA_PRESSURE && skill_id != CR_REFLECTSHIELD )
+	if( d_tbl && check_distance_bl(target, d_tbl, sc->data[SC_DEVOTION]->val3) &&
```
Comment (lighta) [inline src/map/battle.c]: possible segfault on sc->data. why does the sc check was removed here?

**Comment by lighta** [comment]
added and clean in that pull request : #731
@Lemongrass3110 why do some mapname was string from 32 to 15 name lenght ?

**Pull request merged**